lib/slack.atd

```diff
@@ -165,3 +165,37 @@ type ('ok, 'err) http_response <ocaml predef module="Result" t="t"> = [
 
 type 'ok slack_response = ('ok, string) http_response
   <json adapter.ocaml="Atd_adapters.Slack_response_adapter">
+
+type upload_url_res = {
+  upload_url: string;
+  file_id: string;
+}
+
+type file_v2 = {
+  id: string;
+  ?title: string nullable;
+}
+
+type files_v2 = file_v2 list
+
+type complete_upload_external_req = {
+  files: files_v2;
+  ?channel_id: channel_id nullable;
+  ?thread_ts: timestamp nullable;
+  ?initial_comment: string nullable;
+}
+
+type complete_upload_external_file_res = {
+  url_private: string;
+  url_private_download: string;
+  permalink: string;
+  permalink_public: string;
+}
+
+type complete_upload_external_res = {
+  files: files_v2;
+}
```

Comment:
the `_v2` thing is an implementation detail of the ocaml slack lib or the sdks we've looked into. It doesn't belong here